### DISCUSSION TITLE
Path Replay Backpropagation with inverse caustic rendering tutorial

### DISCUSSION THREAD

**Discussion by griffinandrew** [mitsuba-renderer/mitsuba3#1110] [Q&A] [answered]
Hello,

I am curious if it is possible to implement PRB with the inverse caustic rendering tutorial https://mitsuba.readthedocs.io/en/latest/src/inverse_rendering/caustics_optimization.html. I understand that the tutorial makes use of a particle tracer and says that it would be difficult or impossible to sample correctly with a standard path tracer. Why is this the case, or what's the high-level intuition behind this idea? Why does the emission of a single direction of light matter here? In addition, how does this affect whether or not PRB can be used in this case? Any help you could provide would be greatly appreciated!

Thank you!!

**Comment by njroussel** [comment, marked as answer]
Hi @griffinandrew 

In that tutorial, we are using a perfect `dielectric`: for a given incident direction the ray will be reflected/refracted into a single & unique direction. In the documentation you'll see this behavior referred to as the BSDF having a dirac Delta distribution. This means that if you start tracing rays from your sensor, the direction in which you exit the glass slab is entirely dictated by how you entered it. The actual problem lies after you've exited the slab, during the last segment of the light path, because the direction with which you reached the emitter will statistically never be the same direction as the emission direction. So you'd never be able to build a light path that has non-zero contribution.
I
n contrast, with a `roughdielectric` you'd always be able to build a valid light path to the emitter when exiting the light slab because the distribution of exiting directions is smooth and not a dirac Delta.This allows you to sample a valid direction to the emitter directly. This technique is called next event estimation. 

With light tracing (`ptracer`), you get a similar effect because the last light segment (from the wall to the sensor) can also use next event estimation because the wall's BSDF's distribution is also smooth (`diffuse`).

This doesn't specifically apply to `prb` but rather to all path tracers.

**Reply by griffinandrew** [reply]
Thanks so much for your help! So my understanding is that the material constraints of the problem make it impossible to properly sample the lightpath and therefore "replay" or perform PRB, is that correct? Thanks a lot for your time and help!

**Reply by njroussel** [reply]
Exactly, there simply wouldn't be any valid lightpaths to apply the "replay"-mechanism onto.
